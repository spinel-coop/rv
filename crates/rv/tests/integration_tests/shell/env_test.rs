use crate::common::RvTest;
use insta::assert_snapshot;

#[test]
fn test_shell_env_succeeds() {
    let test = RvTest::new();
    let output = test.rv(&["shell", "env", "zsh"]);

    assert_snapshot!(output.normalized_stdout());
    assert!(output.success());
}

#[test]
fn test_nushell_env_succeeds() {
    let test = RvTest::new();
    let output = test.rv(&["shell", "env", "nu"]);

    assert_snapshot!(output.normalized_stdout());
    assert!(output.success());
}

#[test]
fn test_shell_env_with_path() {
    let mut test = RvTest::new();
    test.env.insert("PATH".into(), "/tmp/bin".into());
    let output = test.rv(&["shell", "env", "zsh"]);

    assert_snapshot!(output.normalized_stdout());
    assert!(output.success());
}

#[test]
fn test_shell_env_clears_ruby_vars() {
    let mut test = RvTest::new();
    test.env.insert("PATH".into(), "/tmp/bin".into());
    test.env.insert("RUBY_ROOT".into(), "/tmp/ruby".into());
    test.env.insert("RUBY_ENGINE".into(), "ruby".into());
    test.env.insert("RUBY_VERSION".into(), "3.4.5".into());
    test.env.insert("RUBYOPT".into(), "--verbose".into());
    let output = test.rv(&["shell", "env", "zsh"]);

    assert_snapshot!(output.normalized_stdout());
    assert!(output.success());
}

#[test]
fn test_shell_env_clear_gem_vars() {
    let mut test = RvTest::new();
    test.env.insert("PATH".into(), "/tmp/bin".into());
    test.env.insert("GEM_ROOT".into(), "/tmp/ruby/gems".into());
    test.env.insert("GEM_HOME".into(), "/tmp/root/.gems".into());
    test.env.insert(
        "GEM_PATH".into(),
        "/tmp/root/.gems/bin:/tmp/ruby/gems".into(),
    );
    let output = test.rv(&["shell", "env", "zsh"]);

    assert_snapshot!(output.normalized_stdout());
    assert!(output.success());
}

#[test]
fn test_shell_env_with_ruby() {
    let mut test = RvTest::new();
    test.create_ruby_dir("ruby-3.3.5");

    test.env.insert("PATH".into(), "/tmp/bin".into());
    test.env.insert("RUBY_ROOT".into(), "/tmp/ruby".into());
    test.env.insert("RUBY_ENGINE".into(), "ruby".into());
    test.env.insert("RUBY_VERSION".into(), "3.4.5".into());
    test.env.insert("RUBYOPT".into(), "--verbose".into());
    test.env.insert("GEM_ROOT".into(), "/tmp/ruby/gems".into());
    test.env.insert("GEM_HOME".into(), "/tmp/root/.gems".into());
    test.env.insert(
        "GEM_PATH".into(),
        "/tmp/root/.gems/bin:/tmp/ruby/gems".into(),
    );

    let output = test.rv(&["shell", "env", "zsh"]);
    output.assert_success();

    assert_snapshot!(output.normalized_stdout());
}
