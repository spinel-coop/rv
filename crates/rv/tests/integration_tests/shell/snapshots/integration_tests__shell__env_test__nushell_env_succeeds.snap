---
source: crates/rv/tests/integration_tests/shell/env_test.rs
expression: output.normalized_stdout()
---
{"GEM_HOME":{},"GEM_PATH":{},"GEM_ROOT":{},"PATH":"","RUBYOPT":{},"RUBY_ENGINE":{},"RUBY_ROOT":{},"RUBY_VERSION":{}}
